homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -342,6 +344,14 @@ class XiaomiMiioSensorDescription(SensorEntityDescription):
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
     ),
+    ATTR_LAST_EVENT: XiaomiMiioSensorDescription(
+        key=ATTR_LAST_EVENT,
```

Comment:
What is the sensor state value for these sensors?

Reply:
It is a string containing the last action/event of that device, so:
move/flip90/flip180/taptap/shakeair/rotate
or
l_click_ch0/click_ch0/d_click_ch0/l_click_ch1/click_ch1/d_click_ch1/both_l_click/both_click/both_d_click
(where ch=channel, l=long, d=double, can not be written fully due to char limit in communication to the device).

Reply:
Of course diffrent types of devices will have diffrent sets of possible values.

Reply:
Maybe disable this entity by default? A device that sends events like that should be exposed through a custom event in Home Assistant. The last event is normally not something we expose as state.

Reply:
I do fire custom events upon receiving the push callback.
It is in line 181-202 of the gateway.py file.
However I do that using a entity (I setup the entity with the last_event sensor and fire the custom events from there.

The devices that have last_event or last_push are buttons that only have the last_event entity. The only thing you can do with those devices is push the button and get the event in HomeAssistant.

I do not think async_added_to_hass will be called when the entity is disabled right?
So how do I register the callback and subscribe to the event withouth setting up a entity?

Reply:
Ok, the entity should not fire the events. That should happen from something that is not an entity platform.

Reply:
But then a wall swith like this: https://www.aqara.com/en/product/wireless-remote-switch-h1
Will not have any entities and therefore also no device inside HomeAssistant.
So a user will have a hard time discovering that such a device is actually supported and used by HomeAssistant.

Futhermore when multiple of the same wall switches are used, how do you distinguish between which switch fired the click_ch0 event? (Now that is done through the entity_id of the entity that fired the event).

Is there some documentation on how to properly integrate wireless wall switches/buttons that as sole purpose have to fire events to trigger other stuff. (In HomeAssistant automations)?

Reply:
Sorry, I found the relevant documentation here:
https://developers.home-assistant.io/docs/integration_events
https://developers.home-assistant.io/docs/device_automation_trigger/

I will start converting this.

Reply:
Alright, everything has been converted to device triggers, including translations and everything is setup in the __init__.py and not in the entities anymore.

The Last Event and Last Press entities are now disabled by default but still fire events and provide divice triggers that can be used in automations.

Reply:
Please remove these entities. We don't use entities for events.

Reply:
Done, I removed them.